packages/ckeditor5-html-support/tests/htmlcomment.js

```diff
@@ -222,6 +223,263 @@ describe( 'HtmlComment', () => {
+		} );
+
+		it( 'should throw an error when a comment with the given comment ID does not exist', () => {
+			editor.setData( '<p>Foo</p><p>Bar</p><p>Baz</p>' );
+
+			editor.model.change( writer => {
+				const root = editor.model.document.getRoot();
+				const position = writer.createPositionAt( root, 1 );
+
+				return htmlCommentPlugin.createHtmlComment( position, 'bar' );
+			} );
+
+			expectToThrowCKEditorError( () => {
+				htmlCommentPlugin.removeHtmlComment( 'invalid-comment-id' );
+			}, /^html-comment-does-not-exist/, null );
+		} );
+	} );
+
+	describe( 'getHtmlCommentsInRange()', () => {
+		it( 'should return all comment marker IDs present in the specified range', () => {
+			editor.setData( '<p>Foo</p><p>Bar</p><p>Baz</p>' );
+
+			htmlCommentPlugin.createHtmlComment(
+				createPositionFromPath( [ 1, 0 ] ),
+				'foo'
+			);
+
+			htmlCommentPlugin.createHtmlComment(
+				createPositionFromPath( [ 2 ] ),
+				'bar'
+			);
+
+			const id3 = htmlCommentPlugin.createHtmlComment(
+				createPositionFromPath( [ 2, 1 ] ),
+				'foo'
+			);
+
+			const id4 = htmlCommentPlugin.createHtmlComment(
+				createPositionFromPath( [ 2, 3 ] ),
+				'foo'
+			);
+
+			const range = editor.model.change( writer => {
+				return writer.createRangeIn( root.getChild( 2 ) );
+			} );
+
+			expect( htmlCommentPlugin.getHtmlCommentsInRange( range ) ).to.deep.equal( [ id3, id4 ] );
+		} );
+
+		it( 'should return all comment marker IDs present in the specified range including comments at range boundaries', () => {
+			editor.setData( '<p>Foo</p><p>Bar</p><p>Baz</p>' );
+
+			htmlCommentPlugin.createHtmlComment(
+				createPositionFromPath( [ 1, 0 ] ),
+				'foo'
+			);
+
+			htmlCommentPlugin.createHtmlComment(
+				createPositionFromPath( [ 2 ] ),
+				'bar'
+			);
+
+			const posStart = createPositionFromPath( [ 2, 1 ] );
+			const posEnd = createPositionFromPath( [ 2, 3 ] );
+
+			const id3 = htmlCommentPlugin.createHtmlComment( posStart, 'baz' );
+			const id4 = htmlCommentPlugin.createHtmlComment( posEnd, 'biz' );
+
+			const range = new Range( posStart, posEnd );
+
+			expect( htmlCommentPlugin.getHtmlCommentsInRange( range ) ).to.deep.equal( [ id3, id4 ] );
+		} );
+
+		it( 'should return all comment marker IDs present in the specified collapsed range', () => {
+			editor.setData( '<p>Foo</p><p>Bar</p><p>Baz</p>' );
+
+			htmlCommentPlugin.createHtmlComment(
+				createPositionFromPath( [ 2, 0 ] ),
+				'foo'
+			);
+
+			htmlCommentPlugin.createHtmlComment(
+				createPositionFromPath( [ 2, 2 ] ),
+				'bar'
+			);
+
+			const position = createPositionFromPath( [ 2, 1 ] );
+
+			const id1 = htmlCommentPlugin.createHtmlComment( position, 'baz' );
+			const id2 = htmlCommentPlugin.createHtmlComment( position, 'biz' );
+
+			const range = new Range( position, position );
+
+			expect( htmlCommentPlugin.getHtmlCommentsInRange( range ) ).to.deep.equal( [ id1, id2 ] );
+		} );
+	} );
+
+	describe( 'getHtmlCommentData()', () => {
```

Comment:
There is a missing test for `getHtmlCommentData()` method if comment does not exist.